src/test-reader/mocha-reader/index.js

```diff
@@ -95,6 +99,94 @@ function applyOnly(rootSuite, eventBus) {
     });
 }
 
+function addLocationToRunnables(inBus, config, runnableOpts) {
+    if (!runnableOpts || !runnableOpts.saveLocations) {
+        return;
+    }
+
+    const sourceMapSupport = tryToRequireSourceMapSupport();
+    const { suiteMethods, testMethods } = getMethodsByInterface(config.ui);
+
+    inBus.on(MochaEventBus.events.EVENT_FILE_PRE_REQUIRE, ctx => {
+        [
+            {
+                methods: suiteMethods,
+                eventName: MochaEventBus.events.EVENT_SUITE_ADD_SUITE,
+            },
+            {
+                methods: testMethods,
+                eventName: MochaEventBus.events.EVENT_SUITE_ADD_TEST,
+            },
+        ].forEach(({ methods, eventName }) => {
+            methods.forEach(methodName => {
+                ctx[methodName] = withLocation(ctx[methodName], { inBus, eventName, sourceMapSupport });
+
+                if (ctx[methodName]) {
+                    ctx[methodName].only = withLocation(ctx[methodName].only, { inBus, eventName, sourceMapSupport });
+                    ctx[methodName].skip = withLocation(ctx[methodName].skip, { inBus, eventName, sourceMapSupport });
+                }
+
+                if (!config.ui || config.ui === "bdd") {
+                    const pendingMethodName = `x${methodName}`;
+                    ctx[pendingMethodName] = withLocation(ctx[pendingMethodName], {
+                        inBus,
+                        eventName,
+                        sourceMapSupport,
+                    });
+                }
+            });
+        });
+    });
+}
+
+function withLocation(origFn, { inBus, eventName, sourceMapSupport }) {
+    if (!_.isFunction(origFn)) {
+        return origFn;
+    }
+
+    const wrappedFn = (...args) => {
+        const origStackTraceLimit = Error.stackTraceLimit;
+        const origPrepareStackTrace = Error.prepareStackTrace;
+
+        Error.stackTraceLimit = 2;
```

Comment:
Use only 2 last messages from stack trace. First message it is this file. And second test file.